tests/test_mariadb.py

```diff
@@ -276,3 +280,108 @@ def test_mariadb_healthcheck_galera_cluster_disabled(auto_container_per_test):
+        base="registry.suse.com/suse/mariadb:10.6",
+        containerfile='RUN set -euo pipefail; head -n -1 /usr/local/bin/gosu > /tmp/gosu;  echo \'exec setpriv --pdeathsig=keep --reuid="$u" --regid="$u" --clear-groups -- "$@"\' >> /tmp/gosu;  mv /tmp/gosu /usr/local/bin/gosu; chmod +x /usr/local/bin/gosu',
+        volume_mounts=mounts,
+        extra_environment_variables=_DB_ENV,
+    )
+    mariadb_new = DerivedContainer(
+        base=ctr_image,
+        volume_mounts=mounts,
+        extra_environment_variables=_DB_ENV,
+    )
+    with ContainerLauncher.from_pytestconfig(
+        mariadb_old, container_runtime, pytestconfig
+    ) as launcher:
+        launcher.launch_container()
+        con = launcher.container_data.connection
+        _wait_for_server(con)
+        mariadb_cmd = f"mariadb --user={_OTHER_DB_USER} --password={_OTHER_DB_PW} --host=0.0.0.0 {_TEST_DB}"
+
+        con.check_output(
+            f'echo "CREATE TABLE random_strings (string VARCHAR(255) NOT NULL);" | {mariadb_cmd}'
+        )
+
+        for _ in range(10):
+            con.check_output(
+            f"echo 'INSERT INTO random_strings (string) VALUES (MD5(RAND()));' | {mariadb_cmd}"
+            )
+
+        rows = (
+            con.check_output(f'echo "SELECT count(*) FROM random_strings;" | {mariadb_cmd}')
+            .strip()
+            .splitlines()
+        )
+
+        assert rows and len(rows) == 2
+        _, count = rows[-1].split()
+        assert count == "10"
+
+        con.check_output(
+            f'echo "CREATE TABLE random_numbers (number INT NOT NULL);" | {mariadb_cmd}'
+        )
+
+        for _ in range(12):
+            con.check_output(
+            f"echo 'INSERT INTO random_numbers (number) VALUES (RAND());' | {mariadb_cmd}"
+            )
+
+        rows = (
+            con.check_output(f'echo "SELECT count(*) FROM random_numbers;" | {mariadb_cmd}')
+            .strip()
+            .splitlines()
+        )
+
+        assert rows and len(rows) == 2
+        _, count = rows[-1].split()
+        assert count == "12"
+
+    with ContainerLauncher.from_pytestconfig(
+        mariadb_new, container_runtime, pytestconfig
+    ) as launcher:
+        launcher.launch_container()
+        con = launcher.container_data.connection
+        _wait_for_server(con)
+        mariadb_cmd = f"mariadb --user={_OTHER_DB_USER} --password={_OTHER_DB_PW} --host=0.0.0.0 {_TEST_DB}"
+        rows = (
+            con.check_output(f'echo "SELECT count(*) FROM random_strings;" | {mariadb_cmd}')
+            .strip()
+            .splitlines()
+        )
+
+        assert rows and len(rows) == 2
+        _, count = rows[-1].split()
+        assert count == "10"
+
+        rows = (
+            con.check_output(f'echo "SELECT count(*) FROM random_numbers;" | {mariadb_cmd}')
+            .strip()
+            .splitlines()
+        )
+
+        assert rows and len(rows) == 2
+        _, count = rows[-1].split()
+        assert count == "12"
```

Comment:
You can append the following cleanup code to fix the file permissions on `tmp_path` for rootless podman:
```Python
        # The tmp_path folder is chown'd to the mariadb user by the mariadb
        # container, podman remaps that to some subuid that our current user has
        # no permission to delete. With podman unshare we enter the user
        # namespace, where we can fix the file permissions so that the cleanup
        # by pytest works.
        # Note that we have to chown to root as root inside the user namespace
        # is our current user
        if PODMAN_SELECTED and geteuid() != 0:
            host.check_output(f"podman unshare chown -R root {tmp_path}")
```

I'd suggest to wrap the whole thing in a `try … except` block though, otherwise this will not get executed on failure. Ideally we'd move this to a separate fixture though

Reply:
added this code in try and finally blocks.